drivers/SmartThings/zigbee-button/src/init.lua

```diff
@@ -35,6 +35,9 @@ local generate_event_from_zone_status = function(driver, device, zone_status, zb
       zb_rx.address_header.src_endpoint.value,
       event)
   end
+  if device:get_component_id_for_endpoint(zb_rx.address_header.src_endpoint.value) ~= "main" then
```

Comment:
This check is outside the `event ~= nil` check.  You can also use the fact that the `emit_event_for_endpoint` function returns the event which would include the component as a part of it https://github.ecodesamsung.com/iot-hub/scripting-engine/blob/master/lua_libs/st/device.lua#L131 and just check that component instead of using this function.  Although using this function might be a bit clearer.